packages/webchannel-wrapper/src/index.d.ts

```diff
@@ -15,23 +15,65 @@
  * limitations under the License.
  */
 
-// TODO(mikelehen): Flesh out proper types (or figure out how to generate via
-// clutz or something).
-export class XhrIo {}
-export const ErrorCode: any;
-export const EventType: any;
+// WARNING: This is not a complete set of types exported by WebChannelWrapper.
+// It is merely meant to support the usage patterns of the Firestore SDK.
+
+export var EventType: {
+  COMPLETE: string;
+};
+
 export namespace WebChannel {
-  export type EventType = any;
-  export const EventType: any;
+  export var EventType: {
+    OPEN: string;
+    CLOSE: string;
+    ERROR: string;
+    MESSAGE: string;
+  };
+}
+
+export var ErrorCode: {
+  NO_ERROR: number;
+  HTTP_ERROR: number;
+  TIMEOUT: number;
+};
+
+export interface Headers {
+  [name: string]: string | number;
```

Comment:
Out of curiosity, why doesn't this cause problems like StringMap did?

Reply:
I opted not to enable renaming for Object literal access. Since all headers are invalid JavaScript identifiers, they are only ever set using the literal access notation (such as `header['X-Google-Foo'] = 42`).